### DISCUSSION TITLE
Highlight doesn't work properly with long lines

### DISCUSSION THREAD

**Discussion by scripthunter7** [github-linguist/linguist#6063] [Q&A] [answered]
Hello!

For some reason highlight doesn't work for long lines: [example line](https://github.com/FiltersHeroes/PolishAnnoyanceFilters/blob/9a59b9c41cf5cfa8faedf570bd74408985f1a542/PPB/uBO_AG/popupy_suplement.txt#L5).
Highlight works fine for the same file in VSCode or NovaLightshow: [NovaLightshow demo](https://novalightshow.netlify.app/?grammar-type=url&grammar=https%3A%2F%2Fraw.githubusercontent.com%2Fameshkov%2FVscodeAdblockSyntax%2Fmaster%2Fsyntaxes%2Fadblock.tmLanguage.json&sample-type=url&sample=https%3A%2F%2Fraw.githubusercontent.com%2FFiltersHeroes%2FPolishAnnoyanceFilters%2F9a59b9c41cf5cfa8faedf570bd74408985f1a542%2FPPB%2FuBO_AG%2Fpopupy_suplement.txt).

What could cause this? Is it a GitHub-specific limitation, or do we need to change something in the highlighter?

----

Reference: https://github.com/uBlockOrigin/uAssets/pull/14695#issuecomment-1241573717 by @krystian3w

**Comment by lildude** [comment, marked as answer]
> Is it a GitHub-specific limitation

Yup. The legacy highlighter that uses the TextMate-compatible grammars like the one for Adblock lists will only process lines up to 1024 chars long.

**Reply by scripthunter7** [reply]
I understand, thanks for the quick reply
